state/machine_internal_test.go

```diff
@@ -0,0 +1,57 @@
+package state
+
+import (
+	gc "gopkg.in/check.v1"
+	"gopkg.in/mgo.v2/bson"
+	"gopkg.in/mgo.v2/txn"
+)
+
+type MachineInternalSuite struct {
+}
+
+var _ = gc.Suite(&MachineInternalSuite{})
+
+func (s *MachineInternalSuite) TestCreateUpgradePrepareLockTxnAssertsMachineAlive(c *gc.C) {
+	arbitraryId := "1"
+	arbitraryData := &upgradeSeriesLock{}
+	for _, op := range createUpgradeSeriesPrepareLockTxnOps(arbitraryId, arbitraryData) {
+		assertVal, ok := op.Assert.(bson.D)
+		if op.C == machinesC && ok && assertVal.Map()["life"] == Alive {
+			c.SucceedNow()
+		}
+	}
+	c.Fatal("Transaction does not assert that machines are of status Alive")
```

Comment:
I have never seen SucceedNow vs Fatal. Does the former prevent the latter from failing the test?
If you want that, why not just "return/break" from inside the loop?

I think I'd usually just do this as:
foundAssertion := false
for _, op := range ... {
  ...
  if blah {
   foundAssert = true
   break
  }
 c.Assert(foundAssert, jc.IsTrue, gc.Commentf("could not find a transaction that asserted the status was alive"))
}